tests/5.1/dispatch/test_dispatch_depend.c

```diff
@@ -0,0 +1,66 @@
+//===---- test_dispatch_depend.c -----------------------------------------===//
+// 
+// OpenMP API Version 5.1
+//
+// Uses dispatch construct as context for variant directive. Uses depend clause
+// which adds depend property to the interoperability requirement set.
+//
+// Inspired by "OpenMP 5.1 Features: The Dispatch Construct" video:
+// https://www.youtube.com/watch?v=ruugaX95gIs
+// 
+//===-------------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int arr[N]; // implicit map array 
+int errors;
+int i = 0;
+int a = 0;
+
+void add_two(int *arr);
+
+#pragma omp declare variant(add_two) match(construct={dispatch})
+void add(int *arr){
+    for (int i = 0; i < N; i++){ // Base function adds 1 to array values
+        arr[i] = i+1;
+    }
+    a = 3;
+}
+
+void add_two(int *arr){
+    OMPVV_TEST_AND_SET_VERBOSE(errors, a != 3);
+    OMPVV_ERROR_IF(errors > 0, "Depend clause on dispatch directive not working properly");
+    for (int i = 0; i < N; i++){
+        arr[i] = i+a; // Variant function adds 3 to array values
+    }
+}
+
+int test_wrapper() { 
+    errors = 0;
+    add(arr);
+    for(i = 0; i < N; i++){
+        OMPVV_TEST_AND_SET_VERBOSE(errors, arr[i] != i+1);
```

Comment:
_Side remark: The 'depend(in: a)' case needs to call something else but 'add' – as the nowait 'add_two' checks for 'arr[i] == i+1' but sets it to 'arr[i] == i+3'; thus, calling 'add_two' again will fail._

Reply:
@tob2 , shouldn't the first call to add call add and not the variant add_two ?

Reply:
Thanks @tob2. I see your concern. The task test is going to replace the base test since we want to ensure that the dependencies are evaluated correctly (and are only considered for the sibling tasks).